CMakeLists.txt

```diff
@@ -38,6 +38,12 @@ if(CMAKE_GENERATOR MATCHES Win64*)
   endif()
 endif()
 
+get_property(GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(NOT CMAKE_BUILD_TYPE AND NOT GENERATOR_IS_MULTI_CONFIG)
+  message(STATUS "Using default build type 'Debug'.")
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Specifies the build type on single-configuration generators." FORCE)
```

Comment:
Is this really a thing?  Do we have a platform for which we only build debug?

Reply:
No. The replaces older code in this file that attempted the same thing poorly. What this does is default to Debug if a build type was not provided and our generator is not a multi config generator (Xcode, VS).